src/Mailgun/Messages/MessageBuilder.php

```diff
@@ -248,7 +248,16 @@ public function addCustomHeader($headerName, $headerData)
         if (!preg_match('/^h:/i', $headerName)) {
             $headerName = 'h:'.$headerName;
         }
-        $this->message[$headerName] = [$headerData];
+
+        if (array_key_exists($headerName, $this->message)) {
+            if (is_array($this->message[$headerName])) {
+                $this->message[$headerName][] = $headerData;
+            } else {
+                $this->message[$headerName] = [$this->message[$headerName], $headerData];
+            }
+        } else {
+            $this->message[$headerName] = $headerData;
+        }
```

Comment:
It should work, but then re-write the PHPUnit asserts will be mandatory, to ensure we can count the number of custom headers (to validate we have the number that we provided), validate the headers keys (adding the `h:` prefix on each one) and the stored values for each header.

Just in case its needed, here's a list of all PHPUnit assertions:
https://phpunit.de/manual/current/en/appendixes.assertions.html